packages/js/src/ai-generator/components/modal-content.js

```diff
@@ -14,51 +14,58 @@ export const ModalContent = () => {
 	const upsellLinkPremium = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell" ), [] );
 	const upsellLinkWooPremiumBundle = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell-woo-seo-premium-bundle" ), [] );
 	const upsellLinkWoo = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell-woo-seo" ), [] );
-	const isPremium = useSelect( select => select( STORE ).getIsPremium(), [] );
-	const isWooSeoUpsell = useSelect( select => select( STORE ).getIsWooSeoUpsell(), [] );
-	const isProduct = useSelect( select => select( STORE ).getIsProduct(), [] );
-	const wooSeoNoPremium = isProduct && ! isWooSeoUpsell && ! isPremium;
-	const isProductCopy = !! ( isWooSeoUpsell || wooSeoNoPremium );
-	const postModalprops = {
-		isProductCopy,
+
+	const isPremiumActive = useSelect( select => select( STORE ).getIsPremium(), [] );
+	const isWooSeoActive = useSelect( select => select( STORE ).getIsWooSeoActive(), [] );
+	const isWooCommerceActive = useSelect( select => select( STORE ).getIsWooCommerceActive(), [] );
+
+	const isProductPost = useSelect( select => select( STORE ).getIsProduct(), [] );
+	const isProductTerm = useSelect( select => select( STORE ).getIsProductTerm(), [] );
+
+	const upsellProps = {
 		upsellLink: upsellLinkPremium,
 	};
 
-	if ( isProductCopy ) {
+	// Use specific copy for product posts.
+	if ( isWooCommerceActive && isProductPost ) {
+		upsellProps.title = __( "Generate product titles & descriptions with AI!", "wordpress-seo" );
+		upsellProps.isProductCopy = true;
+	}
+
+	// Use specific copy for product posts and terms, otherwise revert to the defaults.
+	if ( isWooCommerceActive && ( isProductPost || isProductTerm ) ) {
 		const upsellPremiumWooLabel = sprintf(
 			/* translators: %1$s expands to Yoast SEO Premium, %2$s expands to Yoast WooCommerce SEO. */
 			__( "%1$s + %2$s", "wordpress-seo" ),
 			"Yoast SEO Premium",
 			"Yoast WooCommerce SEO"
 		);
-		postModalprops.newToText = sprintf(
+		upsellProps.newToText = sprintf(
 			/* translators: %1$s expands to Yoast SEO Premium and Yoast WooCommerce SEO. */
 			__( "New in %1$s", "wordpress-seo" ),
 			upsellPremiumWooLabel
 		);
-		postModalprops.title = __( "Generate product titles & descriptions with AI!", "wordpress-seo" );
-		if ( ! isPremium && isWooSeoUpsell ) {
-			postModalprops.upsellLabel = `${ sprintf(
+
+		if ( isPremiumActive ) {
+			upsellProps.upsellLabel = sprintf(
+				/* translators: %1$s expands to Yoast WooCommerce SEO. */
+				__( "Unlock with %1$s", "wordpress-seo" ),
+				"Yoast WooCommerce SEO"
+			);
+			upsellProps.upsellLink = upsellLinkWoo;
+		} else if ( ! isWooSeoActive ) {
+			upsellProps.upsellLabel = `${sprintf(
 				/* translators: %1$s expands to Woo Premium bundle. */
 				__( "Unlock with the %1$s", "wordpress-seo" ),
 				"Woo Premium bundle"
```

Comment:
Not really important, but shouldn't we keep the same order for the translation as in the *_explanation_

So the replacement variable would be "Premium Woo bundle" instead. 

<img width="466" alt="image" src="https://github.com/user-attachments/assets/ba13f22c-f3cd-4927-9cc0-01b75fa0de49">

Reply:
Well the product is called the [Yoast WooCommerce Premium bundle](https://yoast.com/wordpress/plugins/yoast-woocommerce-premium-bundle/), so I would suggest to not change that. Maybe the order should be altered but I don't think it's very confusing either 😅